section_to_course/utils.py

```diff
@@ -0,0 +1,60 @@
+"""
+Utility functions for section_to_course.
+"""
+
+from django.utils import timezone
+
+from cms.djangoapps.contentstore.views.block import update_from_source, duplicate_block
+from xmodule.modulestore.django import modulestore
+from opaque_keys.edx.keys import CourseKey
+from opaque_keys.edx.locator import BlockUsageLocator
+
+from section_to_course.models import SectionToCourseLink
+from xmodule.modulestore.split_mongo import BlockKey
+from xmodule.modulestore.store_utilities import derived_key
+
+
+def paste_from_template(*, destination_course_id, block_id, user):
+    """
+    Copy a block to a destination course.
+
+    Given a source block_id and a destination course id, copy the block to
+    the destination course, overwriting any previous copy of that
+    block in the destination course. It will also copy over all the block's
+    children and any files it determines to be related.
+    """
+    store = modulestore()
+    destination_course_key = CourseKey.from_string(destination_course_id)
+    destination_course = store.get_course(destination_course_key)
+    block_usage_key = BlockUsageLocator.from_string(block_id)
+    block_key = BlockKey(block_usage_key.block_type, block_usage_key.block_id)
```

Comment:
Would it make sense to change the `derived_key` function to accept `BlockUsageLocator` instead of the `BlockKey`? The `_copy_from_template` method in the `edx-platform` already has the `BlockUsageLocator` (and we have it here as the `block_usage_key`), so it doesn't make too much sense to have this additional conversion that only creates a `namedtuple`.

Reply:
I like the symmetry of putting a BlockKey in and getting a derived BlockKey out. It feels cleaner, even though we would save a few lines.